sdk/tests/extra/getBufferSubDataAsync-dom-suspension-manual.html

```diff
@@ -0,0 +1,125 @@
+<!--
+
+/*
+** Copyright (c) 2016 The Khronos Group Inc.
+**
+** Permission is hereby granted, free of charge, to any person obtaining a
+** copy of this software and/or associated documentation files (the
+** "Materials"), to deal in the Materials without restriction, including
+** without limitation the rights to use, copy, modify, merge, publish,
+** distribute, sublicense, and/or sell copies of the Materials, and to
+** permit persons to whom the Materials are furnished to do so, subject to
+** the following conditions:
+**
+** The above copyright notice and this permission notice shall be included
+** in all copies or substantial portions of the Materials.
+**
+** THE MATERIALS ARE PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,
+** EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF
+** MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.
+** IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY
+** CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT,
+** TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE
+** MATERIALS OR THE USE OR OTHER DEALINGS IN THE MATERIALS.
+*/
+
+-->
+
+<!DOCTYPE html>
+<html>
+<head>
+<meta charset="utf-8">
+<title>WebGL getBufferSubDataAsync stress test.</title>
+<link rel="stylesheet" href="../../resources/js-test-style.css"/>
+<script src="../../js/js-test-pre.js"></script>
+<script src="../../js/webgl-test-utils.js"></script>
+</head>
+<body>
+<div id="description"></div>
+<canvas id="canvas" width="1" height="1"></canvas>
+<div id="console"></div>
+<script>
+"use strict";
+description("This test makes sure that getBufferSubDataAsync acts under stress as expected governed by WebGL 2.");
+
+var wtu = WebGLTestUtils;
+var canvas = document.getElementById("canvas");
+
+var gl = wtu.create3DContext(canvas, undefined, 2);
+wtu.glErrorShouldBe(gl, gl.NO_ERROR, "Should be no errors from setup.");
+
+var pbo = gl.createBuffer();
+gl.bindBuffer(gl.PIXEL_PACK_BUFFER, pbo);
+gl.bufferData(gl.PIXEL_PACK_BUFFER, 4, gl.DYNAMIC_READ);
+gl.bindBuffer(gl.PIXEL_PACK_BUFFER, null);
+var readbackBuffer = new Uint8Array(4);
+
+var FRAME_COUNT = 600;
+var frame = 0;
+requestAnimationFrame(render);
+var lastResolvedColor;
+
+function insertTimeoutAssertion() {
+  setTimeout(function() {
+    if (!(readbackBuffer[0] == lastResolvedColor[0] &&
+          readbackBuffer[1] == lastResolvedColor[1] &&
+          readbackBuffer[2] == lastResolvedColor[2] &&
+          readbackBuffer[3] == lastResolvedColor[3])) {
+      testFailed("setTimeout function should always see readbackBuffer == lastResolvedColor: was " +
+          readbackBuffer + " but expected " + lastResolvedColor);
+    }
+  }, 0);
+}
+
+function render() {
+  var TEST_COUNT = 100;
+  var promises = [];
+  debug("kicking off " + TEST_COUNT + " getBufferSubDataAsync calls");
+  for (var i = 0; i < TEST_COUNT; i++) {
+    insertTimeoutAssertion();
+    (function() {
```

Comment:
I actually forgot about these tests and haven't updated them. I need to do this before I merge.